crates/wallet/src/wallet/utils.rs

```diff
@@ -26,9 +26,15 @@ pub trait IsDust {
     fn is_dust(&self, script: &Script) -> bool;
 }
 
+impl IsDust for Amount {
+    fn is_dust(&self, script: &Script) -> bool {
+        *self < script.minimal_non_dust()
+    }
+}
+
 impl IsDust for u64 {
     fn is_dust(&self, script: &Script) -> bool {
-        *self < script.minimal_non_dust().to_sat()
+        Amount::from_sat(*self).is_dust(script)
     }
 }
```

Comment:
nit: I guess this could be moved to `coin_selection`, it's the only place still using `is_dust`   with a `u64`, right ?